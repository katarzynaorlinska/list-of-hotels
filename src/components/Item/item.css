.list__photo{
    width: 150px;
    height: 100px;
    padding: 10px;
}

.list{
    list-style: none;
    height: 20vh;
    background-color: white;
    margin-bottom: 10px;
    display: flex;
}

.list__rightBox p{
    display: block;
}