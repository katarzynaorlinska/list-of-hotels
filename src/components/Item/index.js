import "./item.css";

function Item(props) {
  return (
    <li key={props.id}>
      {props.name} {props.city}
    </li>
  );
}

export default Item;
