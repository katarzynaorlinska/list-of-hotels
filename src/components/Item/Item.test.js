import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import mockResponse from "../../mock/listHotel.json";
import Item from "./index";

jest.mock("react-router-dom", () => ({
  Link: () => "Link",
}));

jest.mock("@fortawesome/react-fontawesome", () => ({
  FontAwesomeIcon: () => "FontAwesomeIcon",
}));

test("Render active stars", () => {
  const { container } = render(<Item {...mockResponse.hotels[0]} />);
  expect(container.getElementsByClassName("list__content--active").length).toBe(
    4
  );
});

test("renders correctly", () => {
  const tree = create(<Item {...mockResponse.hotels[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
