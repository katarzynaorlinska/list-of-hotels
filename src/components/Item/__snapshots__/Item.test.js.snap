// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="list"
>
  <div
    className="list__leftBox"
  >
    <img
      alt="hotel"
      className="list__photo"
      src="https://aff.bstatic.com/images/hotel/max500/102/102202664.jpg"
    />
  </div>
  <div
    className="list__rightBox"
  >
    Link
    <p
      className="list__content--city"
    />
    <div
      className="list__content--rate"
    >
      Rate: 
      <div
        className="list__content--active"
      >
        FontAwesomeIcon
      </div>
      <div
        className="list__content--active"
      >
        FontAwesomeIcon
      </div>
      <div
        className="list__content--active"
      >
        FontAwesomeIcon
      </div>
      <div
        className="list__content--active"
      >
        FontAwesomeIcon
      </div>
      <div
        className="list__content--off"
      >
        FontAwesomeIcon
      </div>
    </div>
  </div>
</li>
`;
